src/transformers/models/rag/modeling_rag.py

```diff
@@ -904,9 +930,37 @@ def generate(
                 output_sequences = torch.stack(list({str(k.tolist()): k for k in output_sequences}.values()))
 
             # then, run model forwards to get nll scores:
-            new_input_ids = input_ids[index : index + 1].repeat(len(output_sequences), 1)
-            outputs = self(new_input_ids, labels=output_sequences, exclude_bos_score=True)
-            top_cand_inds = (-outputs["loss"]).topk(num_doc_return_sequences)[1]
+            if input_ids is not None:
+                new_input_ids = input_ids[index : index + 1].repeat(len(output_sequences), 1)
+                outputs = self(new_input_ids, labels=output_sequences, exclude_bos_score=True)
+                top_cand_inds = (-outputs["loss"]).topk(num_doc_return_sequences)[1]
```

Comment:
can we put this line out of the `if-else` clause and only run it once afterward?